plugins/modules/foreman_location.py

```diff
@@ -131,29 +118,49 @@ def main():
         supports_check_mode=True,
     )
 
-    (entity_dict, state) = module.parse_params()
+    entity_dict = module.clean_params()
 
     module.connect()
 
-    name_or_title = entity_dict.pop('name')
-    parent = entity_dict.pop('parent', None)
+    # Hack for broken foreman api
+    _location_create = next(x for x in module.foremanapi.apidoc['docs']['resources']['locations']['methods'] if x['name'] == 'create')
```

Comment:
We'd need the same for `update` ;)

Reply:
also, I think this should be inside `foreman_helper.py` so that other modules also benefit from it.